src/main/java/org/musicbrainz/search/solrwriter/MBXMLWriter.java

```diff
@@ -314,13 +314,31 @@ private static void adjustScore(float maxScore, Object object,
 		}
 	}
 
+    private void writeResponse(Writer writer, MetadataListWrapper metadatalistwrapper)
+            throws IOException {
+        StringWriter sw = new StringWriter();
+        try {
+            marshaller.marshal(metadatalistwrapper.getCompletedMetadata(), sw);
+        } catch (JAXBException e) {
+            e.printStackTrace();
+            return;
+        }
+        writer.write(sw.toString());
+
+    }
+
 	public void write(Writer writer, SolrQueryRequest req, SolrQueryResponse res)
 			throws IOException {
 		MetadataListWrapper metadatalistwrapper = new MetadataListWrapper();
 
 		NamedList vals = res.getValues();
 
 		ResultContext con = (ResultContext) vals.get("response");
+		if (con == null)
+		{
```

Comment:
The old search server has count and offset both set to 0 in this case (http://search.musicbrainz.org/ws/2/area/?query=Area:Berlin), so you might need an additional call to `metadatalistwrapper.setCountAndOffset(0, 0);` here.